src/renderers/WebGLRenderer.d.ts

```diff
@@ -223,8 +223,9 @@ export class WebGLRenderer implements Renderer {
 
   /**
    * Sets the viewport to render from (x, y) to (x + width, y + height).
+   * (x, y) is the lower-left corner of the region.
    */
-  setViewport(x?: number, y?: number, width?: number, height?: number): void;
+  setViewport(x: Vector4 | number, y?: number, width?: number, height?: number): void;
```

Comment:
This looks good. `|` is the ` union type` so both `Vector4` and `number` are accepted. 

> Now, there must be either one arg, or four args.

I don't know if it's possible to define this with typescript since `y`, `width` and `height` are still optional. If you just look at the signature, it would be valid to invocate the method like so:
```js
renderer.setViewport( 0, 0, width );
```

Reply:
Thanks, @Mugen87 . I agree with all that. I am not going to worry about it.